app/grandchallenge/evaluation/forms.py

```diff
@@ -157,12 +161,19 @@ def __init__(self, *args, **kwargs):
         self.fields["parent"].queryset = self.instance.parent_phase_choices
         self.fields["evaluation_requires_memory_gb"].validators = [
             MinValueValidator(settings.ALGORITHMS_MIN_MEMORY_GB),
-            MaxValueValidator(settings.ALGORITHMS_MAX_MEMORY_GB),
+            MaxValueValidator(self.instance.maximum_settable_memory_gb),
         ]
+        self.fields["evaluation_requires_gpu_type"].widget = forms.Select(
+            choices=[
+                (c.value, c.label)
+                for c in self.instance.get_selectable_gpu_types()
+            ]
+        )
```

Comment:
I'm not sure if this is the way to do it. I wanted to dynamically adjust the choices on the field itself; but  that does not appear to be possible?

Reply:
The following does not work? 

```python
self.fields["evaluation_requires_gpu_type"].choices = [(c.value, c.label) for c in self.instance.get_selectable_gpu_types()]
```

Reply:
This is embarrassing, yes that works. (I really thought I had tried that...) Thanks!